### PR TITLE
Wp8 2.2 update

### PR DESCRIPTION
**Pull request by stammen**
This pull requests adds the following to cocos2d-x 2,2 on WP8 and WinRT
1. Precompiled headers are enabled for much faster builds
2. Angle libs are now precompiled.
3. CURL and libWebSockets are enabled for WP8. Prebuilt libs for curl and libwebsocket were added
4. WP8-XAML uses DrawingSurfaceBackgroundGrid for better performance and to fix Device memory leak with DrawingSurface.
5. removed debug prebuilt libs from winrt and wp8.
6. Added purgeSharedSpriteFrameCache() to purge in CCDirector to fix dangling refcounts on textures.

This pull requests adds the following to cocos2d-x 2,2 on Win32

TestCpp libExtensions and libCocos2d-x now have precompiled headers enabled which greatly speeds up the build,

### CODE REVIEW

**Comment by martell** [comment]
Looks good to me code wise. You even have in the pthread to c++11 stuff @koowolf was working on, Very nice Dale, I badly needed us to switch to prebuilt libs. Build times in vs were becoming a pain. I'll do some testing in the morning and post back my results.

@amarcu5 Let me know how you get on with winphone 8.1:)

@stammen I'll be updating openssl and curl and websockets etc to newer version when I find the time. College is wrapping up soon for the year so I'll be able to do it after that. :)
O and I'm hoping you will lend a hand at solving the WinRT winsock port.

File: cocos2dx/CCDirector.cpp [hunk 1]
```diff
@@ -458,7 +458,11 @@ void CCDirector::purgeCachedData(void)
     CCLabelBMFont::purgeCachedData();
     if (s_SharedDirector->getOpenGLView())
     {
+        CCSpriteFrameCache::sharedSpriteFrameCache()->purgeSharedSpriteFrameCache();
         CCTextureCache::sharedTextureCache()->removeUnusedTextures();
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_WP8)
+        CCTextureCache::sharedTextureCache()->dumpCachedTextureInfo();
```
Comment (minggo) [inline cocos2dx/CCDirector.cpp:464]: I think it is better to dump information in debug mode. And I think all platforms should dump this information.

Comment (stammen) [inline cocos2dx/CCDirector.cpp:464]: How about

#if defined(COCOS2D_DEBUG) && (COCOS2D_DEBUG > 0)
CCTextureCache::sharedTextureCache()->dumpCachedTextureInfo();
#endif

Comment (minggo) [inline cocos2dx/CCDirector.cpp:464]: I think it is reasonable.
And how about the progress on 3.0?

**Comment by martell** [comment]
@minggo after the PR is accepted can the authors and changelog be updated on both branches to reflect the implementation of networking and websockets work and the port of libcurl and libwebsockets to wp8 I done. I need to strengthen my CV in the area of games development :)

File: cocos2dx/CCDirector.cpp (hunk 1 above)
Comment (stammen) [inline cocos2dx/CCDirector.cpp:464]: I pushed the change.

3.0 pull request coming later this week.

**Comment by minggo** [comment]
@martell 
Sure. We will do it as you said.

**Comment by minggo** [comment]
@Dhilan007
Could you please test it. 
Thanks.

**Comment by minggo** [comment]
[ci rebuild]

**Comment by WenhaiLin** [comment]
@minggo 
Yes,I will test it ASAP.

**Comment by martell** [comment]
@minggo I am testing now. Will report back shortly

**Comment by minggo** [comment]
@martell
Thanks.

**Comment by martell** [comment]
@minggo All looks well here.
I built hellocpp and testcpp for arm and x86 in both debug and release and it works fine. I can't find anything wrong with it.

The only strange thing was that luatests was a blank app and doesn't include all the dll's
That's not exactly related to this PR I'm guessing? I think lua on wp8 is a wip atm.

Otherwise all looks good. 
It doesn't break other platforms either as I've built both android and win32 to test that

**Comment by martell** [comment]
@minggo What happened this PR?
Is it not going to be merged?

**Comment by ricardoquesada** [comment]
~~re-opened~~
It was automatically closed when `master` was renamed to `v2`

I tried to re-open but I couldn't since `master` branch is no longer present.

@martell 
Could you re-send the patch to `v2` branch? Thanks!

**Comment by stammen** [comment]
I made a new pull request with this code merged with the cocos2d-x v2 branch

https://github.com/cocos2d/cocos2d-x/pull/6516

**Pull request closed without merging**